libnopegl/src/glsl/helper_blur.glsl

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2024 Matthieu Bouron <matthieu.bouron@gmail.com>
+ * Copyright 2024 Nope Forge
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+vec4 ngli_blur_hexagonal(sampler2D tex, vec2 tex_coord, sampler2D map, vec2 map_coord, vec2 direction, float scale, float lod, int nb_samples)
+{
+    float use_coc = nb_samples > 1 ? 1.0 : 0.0;
+    float offset = 0.5 * use_coc;
+
+    vec2 tex_step = 1.0 / vec2(textureSize(tex, 0).xy);
+    vec2 tex_direction = tex_step * direction;
+
+    tex_coord += tex_direction * offset;
+    tex_direction *= scale;
+
+    vec4 color = vec4(0.0);
+    float amount = 0.0;
+    for (int i = 0; i < nb_samples; i++) {
+        vec2 coord_offset = float(i) * tex_direction;
+        vec4 value = textureLod(tex, tex_coord + coord_offset, lod);
+        float coc = texture(map, map_coord + coord_offset).r;
+        coc = mix(1.0, coc, use_coc);
+        color += vec4(ngli_srgb2linear(value.rgb), value.a) * coc;
+        amount += coc;
+    }
```

Comment:
If `use_coc` is `0`, the texture mapping sampling is irrelevant. Is this on purpose?

Wouldn't it make sense to do something like that?

```glsl
    if (use_coc) {
        for (int i = 0; i < nb_samples; i++) {
            vec2 coord_offset = float(i) * tex_direction;
            vec4 value = textureLod(tex, tex_coord + coord_offset, lod);
            color += vec4(ngli_srgb2linear(value.rgb), value.a);
        }
        amount = float(nb_samples);
    } else {
        for (int i = 0; i < nb_samples; i++) {
            vec2 coord_offset = float(i) * tex_direction;
            vec4 value = textureLod(tex, tex_coord + coord_offset, lod);
            float coc = texture(map, map_coord + coord_offset).r;
            color += vec4(ngli_srgb2linear(value.rgb), value.a) * coc;
            amount += coc;
        }
    }
```

Reply:
This will depends on the asm generated by the shader. I plan to do some profiling and do a separated PR improving the blur shaders performances. For the time behind, I prefer to keep it that way, having separate branches like that is likely to results in both branch behind executed by the GPU.